src/lib.rs

```diff
@@ -1,65 +1,51 @@
-extern crate wasm_bindgen;
-extern crate console_error_panic_hook;
-
-use ed25519_dalek::{Keypair as DalekKeypair, SecretKey, PublicKey, PUBLIC_KEY_LENGTH, SECRET_KEY_LENGTH};
-use std::panic;
+use ed25519_dalek::{Keypair as Ed25519Keypair, PublicKey, SecretKey};
 use rand::rngs::OsRng;
-use wasm_bindgen::prelude::*;
+use std::panic;
+use wasm_bindgen::prelude::wasm_bindgen;
 
 #[wasm_bindgen]
 pub fn set_panic_hook() {
     panic::set_hook(Box::new(console_error_panic_hook::hook));
 }
 
 #[wasm_bindgen]
-pub fn hello() -> String {
-    "Hallo, hier ist alles schön".into()
-}
-
-
-#[wasm_bindgen]
-pub struct PandaKeyPair {
+pub struct KeyPair {
     public: PublicKey,
-    private: SecretKey
+    private: SecretKey,
 }
 
 #[wasm_bindgen]
-impl PandaKeyPair {
+impl KeyPair {
     #[wasm_bindgen(constructor)]
-    pub fn new () -> PandaKeyPair {
+    pub fn new() -> Self {
         let mut csprng: OsRng = OsRng {};
-        let key_pair: DalekKeypair = DalekKeypair::generate(&mut csprng);
-        println!("{:?}", key_pair.to_bytes().to_vec());
-        PandaKeyPair {
+        let key_pair = Ed25519Keypair::generate(&mut csprng);
+
+        Self {
```

Comment:
aha!

Reply:
Yes, here I renamed `PandaKeyPair` to `KeyPair` but I'm open to change it back or to something else.

I'm not sure yet how we should handle naming in general, I like the idea of having more-panda names for repositories but when it comes to classes, variables, methods etc. I prefer to keep it simple especially when the system gets complicated, to not make it even harder for new people to understand what is going on (think the puns disappear after you read them first and then it gets confusing). Maybe we can have a simple guideline / styleguide for naming things?

Reply:
I was commenting because of the use of `Self` for the return value :D I agree with everything else! I originally named it PandaKeyPair because we already had KeyPair in the namespace when I wrote that. Live programming etc etc